xbmc/addons/interfaces/GUI/Window.cpp

```diff
@@ -280,6 +315,578 @@ bool Interface_GUIWindow::do_modal(void* kodiBase, void* handle)
 
   return true;
 }
+//@}
+
+/*!
+ * Window control functions
+ */
+//@{
+bool Interface_GUIWindow::set_focus_id(void* kodiBase, void* handle, int control_id)
+{
+  CAddonDll* addon = static_cast<CAddonDll*>(kodiBase);
+  CGUIAddonWindow* pAddonWindow = static_cast<CGUIAddonWindow*>(handle);
+  if (!addon || !pAddonWindow)
+  {
+    CLog::Log(LOGERROR, "Interface_GUIWindow::%s - invalid handler data (kodiBase='%p', handle='%p') on addon '%s'",
+                          __FUNCTION__, addon, pAddonWindow, addon ? addon->ID().c_str() : "unknown");
+    return false;
+  }
+
+  CGUIWindow *pWindow = dynamic_cast<CGUIWindow*>(g_windowManager.GetWindow(pAddonWindow->m_windowId));
+  if (!pWindow)
+    return false;
+
+  if (!pWindow->GetControl(control_id))
+  {
+    CLog::Log(LOGERROR, "Interface_GUIWindow - %s: %s - Control does not exist in window", __FUNCTION__, addon->Name().c_str());
+    return false;
+  }
+
+  Interface_GUIGeneral::lock();
+  CGUIMessage msg = CGUIMessage(GUI_MSG_SETFOCUS, pAddonWindow->m_windowId, control_id);
+  pWindow->OnMessage(msg);
```

Comment:
Have testet, compiler does not accept to have in OnMessage, I change it to 
```cpp
  CGUIMessage msg(GUI_MSG_SETFOCUS, pAddonWindow->m_windowId, control_id);
  pWindow->OnMessage(msg);
```
to match the others.